### ISSUE TITLE
All systems experiencing a boom

### ISSUE THREAD

**Issue by nullfield** [EDCD/EDDI#878]
## EDDI version in which issue found
3.1.0-b2 standalone

## Steps to reproduce

1. Initiate FSD jump to a star system

## Expected
EDDI states system status

## Observed
EDDI stated that every system is experiencing a boom regardless of actual system status, including systems with no human presence.

## Investigation
I checked the journal log and systems were not listed as boom. Also checked System State report event in the Speech responder and didn't see any obvious errors. Restarting both Elite Dangerous and EDDI seems to have fixed it.

**Comment by Tkael** [comment]
Are you using the default scripts? EDDI doesn't have any reason to report a boom state over any other. 
Was the rest of the information that EDDI reported correct to the systems you were passing through?

**Comment by nullfield** [comment]
Yes, I am using the default scripts. (I'm far too lazy to update those.) The rest of the information did seem to be accurate. I'm certain the system names were correct and it also correctly identified when the population was 0. I'm also fairly certain the issue isn't in the scripting as restarting EDDI seems to have cleared it.

**Comment by nullfield** [comment]
A bit of additional data. Started happening again. I departed LHS 20 for Robigo. EDDI said that every system in route was in a boom state. Picked up passengers for Sirius Atmospherics and departed for Sothis. Every system in route was described as booming. Arrived at Sothis and was told system was in lockdown. Every system on the way back to Robigo was then described as in lockdown.

As best I can tell, any system's factionstate persists until a new state is set.

Here's a line (3089) from the attached EDDI issue log file that shows the system has no population but has a faction state of Boom:

`2018-10-17T00:40:31 [Debug] CargoMonitor:PreHandle Received event {"system":"Wredguia WD-K d8-65","x":-319.15625,"y":10.375,"z":-332.3125,"distance":24.23,"fuelused":4.271171,"fuelremaining":27.728828,"allegiance":"None","faction":null,"factionstate":"Boom","economy":"None","economy2":"None","government":"None","security":"GAlAXY_MAP_INFO_state_anarchy","population":0,"systemAddress":2243793258827,"Economy":{"edname":"$economy_None","basename":"None"},"Economy2":{"edname":"$economy_None","basename":"None"},"Allegiance":{"edname":"$faction_none;","basename":"none"},"raw":"{ \"timestamp\":\"2018-10-17T00:40:45Z\", \"event\":\"FSDJump\", \"StarSystem\":\"Wredguia WD-K d8-65\", \"SystemAddress\":2243793258827, \"StarPos\":[-319.15625,10.37500,-332.31250], \"SystemAllegiance\":\"\", \"SystemEconomy\":\"$economy_None;\", \"SystemEconomy_Localised\":\"None\", \"SystemSecondEconomy\":\"$economy_None;\", \"SystemSecondEconomy_Localised\":\"None\", \"SystemGovernment\":\"$government_None;\", \"SystemGovernment_Localised\":\"None\", \"SystemSecurity\":\"$GAlAXY_MAP_INFO_state_anarchy;\", \"SystemSecurity_Localised\":\"Anarchy\", \"Population\":0, \"JumpDist\":24.230, \"FuelUsed\":4.271171, \"FuelLevel\":27.728828 }","timestamp":"2018-10-17T00:40:45Z","type":"Jumped"}`

Here is the line from my journal where I jumped into this system:
`
{ "timestamp":"2018-10-17T00:40:45Z", "event":"FSDJump", "StarSystem":"Wredguia WD-K d8-65", "SystemAddress":2243793258827, "StarPos":[-319.15625,10.37500,-332.31250], "SystemAllegiance":"", "SystemEconomy":"$economy_None;", "SystemEconomy_Localised":"None", "SystemSecondEconomy":"$economy_None;", "SystemSecondEconomy_Localised":"None", "SystemGovernment":"$government_None;", "SystemGovernment_Localised":"None", "SystemSecurity":"$GAlAXY_MAP_INFO_state_anarchy;", "SystemSecurity_Localised":"Anarchy", "Population":0, "JumpDist":24.230, "FuelUsed":4.271171, "FuelLevel":27.728828 }`

This was the first system I jumped to after leaving Robigo Mines.

[eddi_issue.zip](https://github.com/EDCD/EDDI/files/2485446/eddi_issue.zip)

**Comment by nullfield** [comment]
I decided to debug EDDI with Visual Studio. Downloaded source code and compiled. Ran it with some potentially interesting breakpoints set and jumped from Robigo to Wredguia WD-K d8-65 again. This is either interesting or bizarre. I'm not quite sure yet.

First breakpoint was at JournalMonitor.cs line 171:
`SystemState factionState = SystemState.FromEDName(JsonParsing.getString(data, "FactionState"));`

Since FactionState doesn't exist in the data dictionary, SystemState.FromEDName returns null. Okay thus far. factionState is then passed to the constructor of JumpedEvent in line 180. It's null.

Line 81 in JumpedEvent.cs is:
`this.factionstate = (factionstate ?? SystemState.None).localizedName;`

Since factionstate is null, this.factionstate **should** be assigned the localized name of SystemState.None. I looked this up in the SystemState resource file in EddiDataDefinitions and it is "None" as expected. **However** it returned "Boom". I wasn't able to debug any farther in as Visual Studio did a step-over when I tried. I think it should be calling this next line (ResourceBasedLocalizedEDNames.cs line 43):
`public string localizedName => resourceManager.GetString(basename) ?? fallbackLocalizedName ?? basename;`

So I'm stumped now. Any ideas why localizedName would return "Boom" (and I'm assuming "Lockdown" as well) for SystemState.None? (I'm seriously wondering if something is wrong with my install of Windows...)
